joeynmt/training.py

```diff
@@ -357,6 +390,12 @@ def train_and_validate(self, train_data: Dataset, valid_data: Dataset) \
 
                 if self.stop:
                     break
+
+            if epoch_no % self.checkpoint_epochfreq == 0:
```

Comment:
Consider specifying this in number of batches as well to be consistent with validation and logging frequency and to offer intermediate checkpoints for very large (WMT-scale) data sets.

Reply:
Do you mean to set up another training termination criteria based on the number of batches passed (or similarly number of updates)? When this new criteria is met, a new checkpoint is saved.(?)

Reply:
I think it makes more sense to save after a certain number of batches instead of epochs, because epochs might be very large. And the batches are counted anyway in `steps` :)

Reply:
The behavior of one checkpoint per epoch can still be achieved by setting `save_freq` to data_size/batch_size (at least roughly).